recipes/pybind11/all/test_cmakedeps/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.4)
+project(test_package CXX)
+
+
+find_package(pybind11 REQUIRED)
+
+pybind11_add_module(test_package MODULE test_package.cpp)
+target_link_libraries(test_package PRIVATE pybind11::pybind11)
+set_property(TARGET test_package PROPERTY CXX_STANDARD 11)
+
+enable_testing()
+
+add_test(run_example
+  ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test.py ${CMAKE_CURRENT_BINARY_DIR}/lib)
```

Comment:
Use the same project files (`CMakeLists.txt`) from `test_package`, even if you need some `if` to skip the `conanbuildinfo.cmake`. It is important to check that both generators use the same `find_package` call, same target names,...

Reply:
I don't think that is a good practice:

1) Tricking the layout to include a `../test_package/CMakeLists.txt` is very dirty and not something you will do except for this situation.
2) Introducing an IF in the original `CMakelists.txt` to skip the `conanbuildinfo.cmake` would be very very weird, so you edit that file and you are supposed to know that is shared between several test_packages.
3) When Conan 2.0 is introduced here, we want to remove/rename the test_package folder and it will require extra effort.
4) I get that using the same target name is important, and that can be checked in the review of the PR.

Overall I think that code reuse, far from helping, will cause more trouble and dirty code.